### PR TITLE
Remove Legacy TXM

### PR DESCRIPTION
**Pull request by RyanRHall**
Addresses [#175212633](https://www.pivotaltracker.com/story/show/175212633)

### CODE REVIEW

File: core/services/bulletprooftxmanager/bulletprooftxmanager.go [hunk 1]
```diff
@@ -155,3 +145,49 @@ func saveReplacementInProgressAttempt(store *strpkg.Store, oldAttempt models.Eth
 		return errors.Wrap(tx.Create(replacementAttempt).Error, "saveReplacementInProgressAttempt failed")
 	})
 }
+
+// GetLINKBalance returns the balance of LINK at the given address
+func GetLINKBalance(config orm.ConfigReader, client eth.Client, address gethCommon.Address) (*assets.Link, error) {
```
Comment (samsondav) [inline core/services/bulletprooftxmanager/bulletprooftxmanager.go:150]: Is this used anywhere>

File: core/store/presenters/presenters.go [hunk 2]
```diff
@@ -92,7 +93,7 @@ func getBalance(store *store.Store, account accounts.Account, balanceType reques
 		}
 		return (*assets.Eth)(bal), nil
 	case linkRequest:
-		return store.TxManager.GetLINKBalance(account.Address)
+		return bulletprooftxmanager.GetLINKBalance(store.Config, store.EthClient, account.Address)
```
Comment (samsondav) [inline core/store/presenters/presenters.go:96]: Ah, here it is.

File: core/web/user_controller.go [hunk 3]
```diff
@@ -170,7 +170,7 @@ func getETHAccount(ctx *gin.Context, store *store.Store, account accounts.Accoun
 		return presenters.ETHKey{}
 	}
 
-	linkBalance, err := txm.GetLINKBalance(account.Address)
+	linkBalance, err := bulletprooftxmanager.GetLINKBalance(store.Config, store.EthClient, account.Address)
```
Comment (samsondav) [inline core/web/user_controller.go:173]: Would it do better to move this method onto the EthClient?

**Review by samsondav — APPROVED** [review]

File: core/web/user_controller.go (hunk 3 above)
Comment (RyanRHall) [reply core/web/user_controller.go:173]: I think that's definitely a better place for this. Thanks.

Comment (j16r) [inline core/web/user_controller.go:173]: could we just use txm in the codebase?

**Review by j16r — COMMENTED** [review]
What is the comparitive code coverage after this PR?

**Comment by RyanRHall** [comment]
@j16r coverage should be very comparable. We already did the work of going through the test suite and converting everything to the BPTXM [here](https://github.com/smartcontractkit/chainlink/pull/3508). Only a handful of tests were entirely removed when they didn't make sense given the new tx schema.

File: core/web/user_controller.go (hunk 3 above)
Comment (RyanRHall) [reply core/web/user_controller.go]: a very good call - I'll make that change before merging

**Comment by j16r** [comment]
@RyanRHall sure but could we compare absolute numbers? i.e. run `-cover` and compare?

**Comment by RyanRHall** [comment]
## Coverage Changes

| file | change | prev | current |
|--|--|--|--|
| core/adapters | decreases | 81.9% | 81.3% |
| core/cmd | decreases | 51.1% | 50.5% |
| internal/cltest | increases | 1.9% | 2% |
| core/services | increases | 85.7% | 86.2% |
| services/bulletprooftxmanager | increases | 71.1% | 72.6% |
| services/chainlink | increases | 77.5% | 77.7% |
| services/eth | decreases | 70.4% | 69.7% |
| core/store | decreases | 74.4% | 66.4% |
| store/orm | increases | 68.1% | 68.5% |
| store/presenters | increases | 0.6% | 4.6% |
| core/web | increases | 74.7% | 75.0% |


**develop**
ok  	github.com/smartcontractkit/chainlink/core	0.175s	coverage: 28.6% of statements
ok  	github.com/smartcontractkit/chainlink/core/adapters	0.573s	coverage: 81.9% of statements
ok  	github.com/smartcontractkit/chainlink/core/assets	0.027s	coverage: 48.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/cmd	36.018s	coverage: 51.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/internal/cltest	0.100s	coverage: 1.9% of statements
ok  	github.com/smartcontractkit/chainlink/core/internal/gethwrappers	0.043s	coverage: 41.2% of statements
ok  	github.com/smartcontractkit/chainlink/core/logger	0.003s	coverage: 42.3% of statements
ok  	github.com/smartcontractkit/chainlink/core/null	0.003s	coverage: 90.7% of statements
ok  	github.com/smartcontractkit/chainlink/core/services	11.578s	coverage: 85.7% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/bulletprooftxmanager	9.368s	coverage: 71.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/chainlink	0.404s	coverage: 77.5% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/eth	6.606s	coverage: 70.4% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/eth/contracts	0.111s	coverage: 33.3% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/fluxmonitor	76.592s	coverage: 83.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/job	4.004s	coverage: 81.5% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/offchainreporting	14.545s	coverage: 22.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/pipeline	30.465s	coverage: 80.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/postgres	2.759s	coverage: 44.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/signatures/ethdss	0.951s	coverage: 97.6% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/signatures/ethschnorr	0.032s	coverage: 90.6% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/signatures/secp256k1	0.065s	coverage: 86.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/synchronization	82.100s	coverage: 78.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/vrf	24.960s	coverage: 79.3% of statements
ok  	github.com/smartcontractkit/chainlink/core/store	30.135s	coverage: 74.4% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/migrations	9.428s	coverage: 71.4% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/models	0.701s	coverage: 58.7% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/models/ocrkey	0.029s	coverage: 39.4% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/models/vrfkey	0.122s	coverage: 50.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/orm	5.914s	coverage: 68.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/presenters	0.077s	coverage: 0.6% of statements
ok  	github.com/smartcontractkit/chainlink/core/utils	10.100s	coverage: 55.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/web	12.012s	coverage: 74.7% of statements

**this branch**
ok  	github.com/smartcontractkit/chainlink/core	0.193s	coverage: 28.6% of statements
ok  	github.com/smartcontractkit/chainlink/core/adapters	0.446s	coverage: 81.3% of statements
ok  	github.com/smartcontractkit/chainlink/core/assets	(cached)	coverage: 48.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/cmd	18.664s	coverage: 50.5% of statements
ok  	github.com/smartcontractkit/chainlink/core/internal/cltest	0.093s	coverage: 2.0% of statements
ok  	github.com/smartcontractkit/chainlink/core/internal/gethwrappers	(cached)	coverage: 41.2% of statements
ok  	github.com/smartcontractkit/chainlink/core/logger	(cached)	coverage: 42.3% of statements
ok  	github.com/smartcontractkit/chainlink/core/null	(cached)	coverage: 90.7% of statements
ok  	github.com/smartcontractkit/chainlink/core/services	7.361s	coverage: 86.2% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/bulletprooftxmanager	14.218s	coverage: 72.6% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/chainlink	1.505s	coverage: 77.7% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/eth	6.592s	coverage: 69.7% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/eth/contracts	0.116s	coverage: 33.3% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/fluxmonitor	80.187s	coverage: 83.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/job	3.962s	coverage: 81.5% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/offchainreporting	17.056s	coverage: 22.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/postgres	2.555s	coverage: 44.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/signatures/ethdss	(cached)	coverage: 97.6% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/signatures/ethschnorr	(cached)	coverage: 90.6% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/signatures/secp256k1	(cached)	coverage: 86.1% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/synchronization	81.118s	coverage: 78.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/services/vrf	26.064s	coverage: 79.3% of statements
ok  	github.com/smartcontractkit/chainlink/core/store	13.030s	coverage: 66.4% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/migrations	6.829s	coverage: 71.4% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/models	1.046s	coverage: 58.7% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/models/ocrkey	0.095s	coverage: 39.4% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/models/vrfkey	(cached)	coverage: 50.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/orm	6.111s	coverage: 68.5% of statements
ok  	github.com/smartcontractkit/chainlink/core/store/presenters	0.098s	coverage: 4.6% of statements
ok  	github.com/smartcontractkit/chainlink/core/utils	10.127s	coverage: 55.8% of statements
ok  	github.com/smartcontractkit/chainlink/core/web	12.454s	coverage: 75.0% of statements

**Review by j16r — APPROVED** [review]

**Pull request merged**